### PR TITLE
Support Quake2 ReRelease

### PR DESCRIPTION
**Pull request by 0lvin**
Fixed for support ReRelease path changes #1039 

Use PNG as replace of pcx automatically, generate colormap and 16to8 if information found(mostly required for soft render)

Rework of nolerp filtering, logic are same as for haptic feedback

Vulkan support https://github.com/yquake2/ref_vk/pull/46

### CODE REVIEW

File: src/client/refresh/soft/sw_image.c [hunk 1]
```diff
@@ -708,8 +711,57 @@ R_InitImages (void)
...
+					for (i = 0; i < 256; i ++)
+					{
+						int r_c, g_c, b_c;
+						float curr_diff;
+
+						r_c = d_8to24table[i * 4 + 0] - (r << 3);
+						g_c = d_8to24table[i * 4 + 1] - (g << 2);
+						b_c = d_8to24table[i * 4 + 2] - (b << 3);
+
+						curr_diff = sqrt(
```
Comment (turol) [inline src/client/refresh/soft/sw_image.c:743]: There's no need to sqrt here since you only care about their relative values and not the actual result.

Comment (0lvin) [reply src/client/refresh/soft/sw_image.c]: Thank you, fixed

**Comment by 0lvin** [comment]
Looks like change is enough to run without critical errors.

Known bugs in render code:
* dark/no light in soft render
* bright lighting in GL1/GL3/VK renders (looks like no shadows at all)
* Slow rotating of sky

@Yamagi could you please review?

**Comment by 0lvin** [comment]
I will port lights fixes from https://github.com/yquake2/ref_vk/pull/47 for map when I will found way to fix light for models.

**Comment by 0lvin** [comment]
I have added additional small fix for palette [r_retexturing: use rgb with pcx if image has builtin palette](https://github.com/yquake2/yquake2/pull/1040/commits/eb89968211876c9b4c805c169778562a38d7e58f).

It resolved case when image is palette based but has different palette than default one:

Before with Daikatana colormap:
daikatana_colormap

Before with Quake2 colormap:
quake2_colormap

After fix:
reuse_colormap

**Comment by 0lvin** [comment]
@Yamagi @DanielGibson What do you think about add Daikatana image format? It will take ~ 211 lines of code change. Look to https://github.com/0lvin/yquake2/tree/rerelease-combined as example.

**Comment by Yamagi** [comment]
Moving home is done and I'm back in the game :) I will look into this into this in the next few days.

**Comment by 0lvin** [comment]
Thank you, I have created one additional #1055. It's not finished.

Code has only GL1 implementations, I have tested code with base1 and mguhub. It has no regression for old maps as I know.

**Comment by Yamagi** [comment]
I have created a new repository for the remastered version: https://github.com/yquake2/yquake2remaster I think it's better to keep the remastered support in a fork until we either have developed it to a point were we can decide if it can be merged back into YQ2 itself or should stay a fork. This way we don't need to be too overcautious with breaking things.

I have already merged this PR into the new repo. @0lvin, I'll give you write access to the new repository.

**Comment by 0lvin** [comment]
Thank you! 

I will merge ref_vk to it also, just to have single source repository without additional branch in ref_vk for ReRelease with autorotate parameter in API.

**Pull request closed without merging**